runtime/parachains/src/inclusion.rs

```diff
@@ -255,20 +262,24 @@ impl<T: Trait> Module<T> {
 		let threshold = availability_threshold(validators.len());
 
 		let mut freed_cores = Vec::with_capacity(n_bits);
-		for (para_id, pending_availability) in assigned_paras_record.into_iter()
+		for (para_id, pending_availability, commitments) in assigned_paras_record.into_iter()
 			.filter_map(|x| x)
-			.filter_map(|(id, p)| p.map(|p| (id, p)))
+			.filter_map(|(id, p, c)| p.and_then(|p| c.map(|c| (id, p, c))))
 		{
 			if pending_availability.availability_votes.count_ones() >= threshold {
 				<PendingAvailability<T>>::remove(&para_id);
+				<PendingAvailabilityCommitments>::remove(&para_id);
```

Comment:
So yeah: we should read the `PendingAvailabilityCommitments` here only and use `take` to load it into memory. Then we can construct a `CommittedCandidateReceipt` to send off to the `enact_candidate` function.

If is is `None`, I would prefer to defensively `continue` because it is possible that governance or something might mess around with storage, in which case we don't want to brick the chain by introducing an unconditional `panic`.

Reply:
> If it is None, I would prefer to defensively continue because it is possible that governance or something might mess around with storage, in which case we don't want to brick the chain by introducing an unconditional panic.

So my question is: are updates to `PendingAvailability` and `PendingAvailabilityCommitments` supposed to be done in a lockstep (atomically). Is it e.g. true that this invriant holds:
```rust
<PendingAvailability<T>>::remove(&para_id).is_some() iff
<PendingAvailabilityCommitments>::remove(&para_id).is_some()
```
As you say, other modules might write to the storage in the meantime creating racing problems or is it not supposed to happen?

Reply:
The invariant should hold and would hold in isolation, but I wouldn't risk the entire chain on it, where the storage can be mangled by anyone in control of a `Root` origin. If governance intervenes to break the storage, it can logically intervene to correct the damage done.